### PR TITLE
Add `ops.random.shuffle`

### PR DESCRIPTION
**Pull request by james77777778**
I believe this op could be helpful when I came across the following PR:
https://github.com/keras-team/keras-core/pull/884#discussion_r1326753668

### CODE REVIEW

File: keras_core/ops/random_test.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+class RandomTest(testing.TestCase, parameterized.TestCase):
+    def test_shuffle(self):
+        x = np.arange(100).reshape(10, 10)
+
+        # Test axis=0
+        expected = ops.random.shuffle(x)
+
+        self.assertFalse(
+            np.all(x == ops.convert_to_numpy(expected))
+        )  # seems unlikely!
```
Comment (fchollet) [inline keras_core/ops/random_test.py:17]: Unlikely but nevertheless the test will be flaky some % of the time unless it is seeded. I recommend using a seed.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR. Yes, this is a useful op! Please also add it in `keras_core/random/random.py` so we can export it to the public API and document it.

File: keras_core/ops/random_test.py (hunk 1 above)
Comment (james77777778) [reply keras_core/ops/random_test.py]: Done. `seed=0` has been added.

**Comment by james77777778** [comment]
> Please also add it in `keras_core/random/random.py` so we can export it to the public API and document it.

I didn't notice that there is a path for `keras_core/random/random.py`.

The function and corresponding docstring have been added.

**Review by fchollet — APPROVED** [review]
LGTM, thank you for the contribution! The test failure is unrelated indeed, just a flake.

**Pull request merged**